### ISSUE TITLE
Somes wav files generate loud white noise at the end

### ISSUE THREAD

**Issue by Tolriq** [google/ExoPlayer#4724]
Exoplayer 2.8.4

A user reported to me that some wav files generate loud white noise at the end of the playback of each track for about 1 second. (On multiple different devices)

But he was also able to provide me a wav file that triggers that bug at the end for unlimited duration.
The media is 4:16 duration, but after the end loud white noise until stop is pressed.

All can be easily reproduced via the demo player, but since I'm pretty sure the files he sent me are copyrighted don't know how to give you the urls without generating issues :)

**Comment by ojw28** [comment]
I don't think we can do anything to look at this unless we have sample content that reproduces the issue.

**Comment by Tolriq** [comment]
@ojw28 
Well the indirect question was how do I send you the said content without posting copyrighted content on a public place :)

**Comment by ojw28** [comment]
This is already answered in the issue template that you would have seen when filing this issue:

```
### Link to test content
Provide a link to media that reproduces the issue. If you don't wish to post it
publicly, please submit the issue, then email the link to
dev.exoplayer@gmail.com using a subject in the format "Issue #1234".
```

**Comment by Tolriq** [comment]
Oups missed that as bugreport was not relevant to the issue :(

Sorry and mail sent.

**Comment by Tolriq** [comment]
@ojw28 @andrewlewis Did you get the samples?

Is this something that could be fixed or are the wav broken and he should rip them again or convert them?

Just to know what to tell the user (wait or start ripping / converting)

**Comment by ojw28** [comment]
We got the samples, thanks. The issue occurs when a WAV file has non-audio data appended onto the end (I don't know whether it's junk data, or useful metadata or something, but it's not audio). As an example, the first sample you provided has a file length of ~4.9MB, but the audio part of it ends at ~4.2MB. We currently try and play out the remainder of the file as audio too, which is what sounds like white noise. Playback does end eventually (I don't think it's true that the white noise continues indefinitely). Next steps are:

* We should fix this in ExoPlayer. The WAV header provides sufficient information for us to determine where the end of the audio data is located, so by stopping there we can fix the problem.
* As an interim measure, your user should be able to fix the problem by re-encoding the media to not contain whatever the non-audio data is that's currently appended onto the end.

**Comment by Halloway66** [comment]
Hi,

I'm the user who sent the original bug report to Tolriq. I read your comment about ExoPlayer playing non-audio data in the WAV and I tried an experiment:

1. I ripped a CD track to WAV with no tags or cover art embedded. File played OK with no white noise.
2. I converted that WAV to FLAC (with Foobar2000) and added some track information and a cover art image. That file played OK too.
3. I converted the FLAC back to WAV, retaining the tags and image. This one did play white noise at the end. 

As a further experiment, I tried embedding a larger image file into the audio file and the white noise played for longer. So it does look like that it's the embedded tags and image that are being played by ExoPlayer. 

I've already converted my few WAVs to FLACs and that has effectively cured the white noise problem for me.

Thanks and best wishes.
